src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Krivodonova.hpp

```diff
@@ -0,0 +1,995 @@
+ *   object to be passed as an additional argument to the `operator()` (still
+ *   would need to be stored in the DataBox).
+ * - We cannot handle the case where neighbors have more/fewer
+ *   coefficients. In the case that the neighbor has more coefficients we could
+ *   just ignore the higher coefficients. In the case that the neighbor has
+ *   fewer coefficients we have a few choices.
+ * - Having a different number of collocation points in different directions is
+ *   not supported. However, it is straightforward to handle this case. The
+ *   outermost loop should be in the direction with the most collocation points,
+ *   while the inner most loop should be over the direction with the fewest
+ *   collocation points. The highest to lowest coefficients can then be limited
+ *   appropriately again.
+ * - h-refinement is not supported, but there is one reasonably straightforward
+ *   implementation that may work. In this case we would ignore refinement that
+ *   is not in the direction of the neighbor, treating the element as
+ *   simply having multiple neighbors in that direction. The only change would
+ *   be accounting for different refinement in the direction of the neighor,
+ *   which should be easy to add since the differences in coefficients in
+ *   Eq.\f$\ref{eq:krivodonova 3d minmod}\f$ will just be multiplied by
+ *   non-unity factors.
+ */
+template <size_t VolumeDim, typename... Tags>
+class Krivodonova<VolumeDim, tmpl::list<Tags...>> {
+ public:
+  /*!
+   * \brief The \f$\alpha_i\f$ values in the Krivodonova algorithm.
+   */
+  struct Alphas {
+    using type = std::array<
+        double, Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>;
+    static constexpr OptionString help = {
+        "The alpha parameters of the Krivodonova limiter"};
+    static type default_value() noexcept {
+      return make_array<
+          Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>(1.0);
+    }
+  };
+  /*!
+   * \brief Turn the limiter off
+   *
+   * This option exists to temporarily disable the limiter for debugging
+   * purposes. For problems where limiting is not needed, the preferred
+   * approach is to not compile the limiter into the executable.
+   */
+  struct DisableForDebugging {
+    using type = bool;
+    static type default_value() noexcept { return false; }
+    static constexpr OptionString help = {"Disable the limiter"};
+  };
+
+  using options = tmpl::list<Alphas, DisableForDebugging>;
+  static constexpr OptionString help = {
+      "The hierarchical slope limiter of Krivodonova.\n\n"
+      "This slope limiter works by limiting the highest modal "
+      "coefficients/derivatives using an aggressive minmod approach, "
+      "decreasing in modal coefficient order until no more limiting is "
+      "necessary."};
+
+  explicit Krivodonova(
+      std::array<double,
+                 Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>
+          alphas,
```

Comment:
You could default this argument to `Alphas::default_value()` if you like

Reply:
I agree with this reducing text but I don't agree with the conceptual connection it builds. I view the options as being a layer on top of the core class and so the class shouldn't rely on the options :)